File: lib/data/non/input/keyboard.rb
class Keyboard
    java_import 'com.badlogic.gdx.Gdx'
    java_import 'non.Non'
    
    def show
        setVisible(true)
    end
    
    def hide
        setVisible(false)
    end
    
    def down(key)
        keycode = Non.getKey(name)
        Gdx.input.isKeyPressed(keycode) && keycode != 0
    end
end

File: lib/data/non/lua/input/keyboard.lua
non.keyboard = {}

local Gdx = NON_GDX
local Non = NON_NON
  
function non.keyboard.show()
  Gdx.input:setOnscreenKeyboardVisible(true)
end
  
function non.keyboard.hide()
  Gdx.input:setOnscreenKeyboardVisible(false)
end
  
function non.keyboard.down(key)
  local keycode = Non:getKey(name)
  if Gdx.input:isKeyPressed(keycode) and keycode ~= 0 then return true end
  return false
end